android/src/main/java/com/reactlibrary/JudoReactNativeModule.kt

```diff
@@ -0,0 +1,669 @@
+            3 -> PaymentWidgetType.CHECK_CARD
+            4 -> PaymentWidgetType.SAVE_CARD
+            else -> PaymentWidgetType.CARD_PAYMENT
+        }
+    }
+
+    private fun getJudoId(options: ReadableMap): String? {
+
+        val configuration = try {
+            options.getMap("configuration")
+        } catch (_: Exception) {
+            return null
+        }
+
+        return try {
+            configuration?.getString("judoId")
+        } catch (_: Exception) {
+            null
+        }
+    }
+
+    private fun getAmount(options: ReadableMap): Amount? {
+
+        val configuration = try {
+            options.getMap("configuration")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val amount = try {
+            configuration?.getMap("amount")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val amountValue = try {
+            amount?.getString("value")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val currencyValue = try {
+            amount?.getString("currency")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val currency = when (currencyValue) {
+            null -> Currency.GBP
+            else -> Currency.valueOf(currencyValue)
+        }
+
+        return try { Amount.Builder()
+                .setAmount(amountValue)
+                .setCurrency(currency)
+                .build() } catch (_: Exception) {
+            null
+        }
+    }
+
+    private fun getReference(options: ReadableMap): Reference? {
+
+        val configuration = try {
+            options.getMap("configuration")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val reference = try {
+            configuration?.getMap("reference")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val consumerReference = try {
+            reference?.getString("consumerReference")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val paymentReference = try {
+            reference?.getString("paymentReference")
+        } catch (_: Exception) {
+            null
+        }
+
+        val metadata = try {
+            reference?.getMap("metadata")
+        } catch (_: Exception) {
+            null
+        }
+
+        var builder = Reference.Builder()
+                .setConsumerReference(consumerReference)
+                .setPaymentReference(paymentReference)
+
+        metadata?.let {
+            val bundle = Bundle()
+            metadata.toHashMap().forEach {
+                bundle.putString(it.key, it.value.toString())
+            }
+            builder = builder.setMetaData(bundle)
+        }
+
+        return try {
+            builder.build()
+        } catch (_: Exception) {
+            null
+        }
+    }
+
+    private fun getSiteId(options: ReadableMap): String? {
```

Comment:
to make this file shorter, let's extract this stuff to extension functions, ex:
```
val Any.configuration: Map<String, Any>?
    get() = try {
        getMap("configuration")
    } catch (_: Exception) {
        null
    }

val Any.siteId: String?
    get() = try {
    configuration?.getString("siteId")
} catch (_: Exception) {
    null
}

```

Reply:
Fixed